client/src/main.rs

```diff
@@ -1460,10 +1460,17 @@ fn listen(matches: &ArgMatches<'_>) {
 						},
 						Event::System(ee) => match ee {
 							frame_system::Event::ExtrinsicFailed {
-								dispatch_error,
-								dispatch_info,
+								dispatch_error: _,
+								dispatch_info: _,
 							} => {
-								error!("ExtrinsicFailed: {:?} {:?}", dispatch_error, dispatch_info);
+								let ed = EventsDecoder::new(api.metadata.clone());
+								let event_records = vec![evr];
+
+								let decoded_event = &ed
+									.decode_events(&mut event_records.encode().as_slice())
+									.unwrap()[0];
+
+								error!("ExtrinsicFailed: {:?}", decoded_event);
```

Comment:
Very cool! I think this was not that easy to find out how it works, I am impressed!

Sorry, for making slight readability adjudments here. :)